src/halmos/__main__.py

```diff
@@ -459,57 +513,47 @@ def run_target_contract(
             continue
 
         try:
-            # initialize symbolic execution environment
-            sevm = SEVM(args, fun_info)
-            solver = mk_solver(args)
-            path = Path(solver)
-            path.extend_path(ex.path)
-
-            # prepare calldata and dynamic parameters
-            cd, dyn_params = mk_calldata(
-                abi, fun_info, args, new_symbol_id=ex.new_symbol_id
-            )
-            path.process_dyn_params(dyn_params)
-
             # create a symbolic tx.origin
             tx_origin = mk_addr(
-                f"tx_origin_{id_str(addr)}_{uid()}_{ex.new_symbol_id():>02}"
+                f"halmos_tx_origin_{id_str(addr)}_{uid()}_{ex.new_symbol_id():>02}"
             )
 
             # create a symbolic msg.sender
             msg_sender = mk_addr(
-                f"msg_sender_{id_str(addr)}_{uid()}_{ex.new_symbol_id():>02}"
+                f"halmos_msg_sender_{id_str(addr)}_{uid()}_{ex.new_symbol_id():>02}"
+            )
+
+            # restrict msg.sender to the specified target senders
+            msg_sender_cond = (
+                Or([msg_sender == target_sender for target_sender in target_senders])
+                if target_senders
+                else None
             )
```

Comment:
we can avoid creating an Or expression with target_senders is a single element:

```suggestion
            msg_sender_cond = (
                msg_sender == target_senders[0]
                if len(target_senders) == 1
                else Or([msg_sender == sender for sender in target_senders])
                if target_senders
                else None
            )
```

Reply:
https://github.com/a16z/halmos/pull/487/commits/ca4f6b3a0da4cc20adcfa7774fdc6f7c809332c7